scripts/generate_service.go

```
@@ -0,0 +1,305 @@
+		ClientType:   clientType,
+	}
+
+	// eg. Address -> Addresses instead of Address -> Addresss
+	if strings.HasSuffix(templateData.ClientType, "s") {
+		templateData.ServiceType = normalisedData.Type + "es"
+	} else {
+		templateData.ServiceType = normalisedData.Type + "s"
+	}
+
+	// Add the strings required for each version that the resource supports.
+	for _, version := range OrderedComputeApiVersions {
+		clientNameSuffix := ""
+		serviceName := "v1"
+
+		if version == v0beta {
+			clientNameSuffix = "Beta"
+			serviceName = "v0beta"
+		}
+
+		templateData.Versions = append(templateData.Versions, VersionInfo{ClientNameSuffix: clientNameSuffix, ServiceName: serviceName})
+		if version == normalisedData.LowestVersion {
+			break
+		}
+	}
+
+	// If a resource can be updated, add information about how to update it
+	if normalisedData.UpdateVersion != none {
+		for _, version := range OrderedComputeApiVersions {
+			clientNameSuffix := ""
+			serviceName := "v1"
+
+			if version == v0beta {
+				clientNameSuffix = "Beta"
+				serviceName = "v0beta"
+			}
+
+			templateData.UpdateVersions = append(templateData.UpdateVersions, VersionInfo{ClientNameSuffix: clientNameSuffix, ServiceName: serviceName})
+			if version == normalisedData.UpdateVersion {
+				break
+			}
+		}
+	}
+
+	// If the resource has extra parameters like a scope or parent resource, add them now.
+	// This works under the assumption that they are always strings.
+	if len(normalisedData.ExtraParams) > 0 {
+		eCall := ""
+
+		for _, param := range normalisedData.ExtraParams {
+			eCall += fmt.Sprintf(" %s,", param)
+		}
+
+		templateData.ExtraParamsCall = eCall
+		templateData.ExtraParamsSignature = eCall[:len(eCall)-1] + " string,"
+	}
+
+	f, err := os.Create(fmt.Sprintf("gen-%s.go", *clientTypeFlag))
+	defer f.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	buf := &bytes.Buffer{}
+	t := template.Must(template.New("temp").Parse(multiversionServiceTemplate))
+	t.Execute(buf, templateData)
```

Ignoring an error here.